janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/NumericTranslationQueryParser.java

```diff
@@ -0,0 +1,175 @@
+        super(field, analyzer);
+        this.storeRetriever = storeRetriever;
+    }
+
+    @Override
+    protected Query newRangeQuery(final String field, final String start, final String end, final boolean startInclusive,
+                                  final boolean endInclusive) {
+
+        Class<?> dataType = storeRetriever.get(field).getDataType();
+        if (Number.class.isAssignableFrom(dataType)) {
+            try {
+                return getNumericRangeQuery(field, dataType, start, end, startInclusive, endInclusive);
+            } catch (NumberFormatException e) {
+                log.warn("Expected Number type for " + field + ":" + dataType + ", but " + e.getMessage());
+            }
+        }
+
+        return super.newRangeQuery(field, start, end, startInclusive, endInclusive);
+    }
+
+    @Override
+    protected Query getRangeQuery(final String field, final String start, final String end, final boolean includeLower,
+                                  final boolean includeUpper) throws ParseException {
+        Class<?> dataType = storeRetriever.get(field).getDataType();
+        if (Number.class.isAssignableFrom(dataType)) {
+            try {
+                return getNumericRangeQuery(field, dataType, start, end, includeLower, includeUpper);
+            } catch (NumberFormatException e) {
+                log.warn("Expected Number type for " + field + ":" + dataType + ", but " + e.getMessage());
+            }
+        }
+
+        return super.getRangeQuery(field, start, end, includeLower, includeUpper);
+    }
+
+    @Override
+    protected Query newTermQuery(Term t) {
+        if (t.field() == null) {
+            return super.newTermQuery(t);
+        }
+
+        Class<?> dataType = storeRetriever.get(t.field()).getDataType();
+        if (Number.class.isAssignableFrom(dataType)) {
+            try {
+                return buildNumericQuery(t.field(), t.text(), dataType);
+            } catch (NumberFormatException e) {
+                log.warn("Expected Number type for " + t.field() + ":" + dataType + ", but " + e.getMessage());
+            }
+        }
+
+        return super.newTermQuery(t);
+    }
+
+    @Override
+    protected Query newWildcardQuery(final Term t) {
+        if (t.field() == null) {
+            return super.newTermQuery(t);
+        }
+
+        Class<?> dataType = storeRetriever.get(t.field()).getDataType();
+        if (Number.class.isAssignableFrom(dataType)) {
+            try {
+                return buildNumericQuery(t.field(), t.text(), dataType);
+            } catch (NumberFormatException e) {
+                log.warn("Expected Number type for " + t.field() + ":" + dataType + ", but " + e.getMessage());
+            }
+        }
+
+        return super.newWildcardQuery(t);
+    }
+
+    private Query getNumericRangeQuery(final String field, final Class<?> type, String start, String end, final boolean includeLower,
```

Comment:
What about other numeric types like `Short` and `Byte`? Will they be supported?

Reply:
Currently LuceneIndex support two number types Long and Double, see function LuceneIndex.addDocument Number case

I'm simplify numeric type parsing, and now Short and Byte supported too